code/game/objects/items/radio/headset.dm

```diff
@@ -349,9 +349,38 @@ GLOBAL_LIST_INIT(channel_tokens, list(
 	freerange = TRUE
 
 
+/obj/item/radio/headset/almayer/marine/Initialize(mapload, squad, rank)
+	if(squad)
+		icon_state = "[squad]_headset"
+		var/dat = "marine [lowertext(squad)]"
+		switch(squad)
+			if("Alpha")
+				frequency = FREQ_ALPHA
+			if("Bravo")
+				frequency = FREQ_BRAVO
+			if("Charlie")
+				frequency = FREQ_CHARLIE
+			if("Delta")
+				frequency = FREQ_DELTA
+		switch(rank)
+			if("Squad Leader")
+				dat += " leader"
+				keyslot2 = new /obj/item/encryptionkey/squadlead
+				use_command = TRUE
+				command = TRUE
+			if("Squad Engineer")
+				dat += " engineer"
+				keyslot2 = new /obj/item/encryptionkey/engi
+			if("Squad Corpsman")
+				dat += " corpsman"
+				keyslot2 = new /obj/item/encryptionkey/med
+		name = dat + " radio headset"
+	return ..()
+
+
 /obj/item/radio/headset/almayer/marine/alpha
 	name = "marine alpha radio headset"
-	icon_state = "sec_headset"
+	icon_state = "Alpha_headset"
```

Comment:
capital names in icon states are usually a bad practice, think it would be easy to change?

Reply:
It is done this way because it's set by `"[squad]_headset"`, with `squad` being capitalized.
It would just require the usage of `[lowertext(squad)]_headset`, which is trivial.
Want it done?

Reply:
Yeah that sounds better